tests/ImageSharp.Tests/TestUtilities/AsyncLocalSwitchableFilesystem.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.IO;
+using System.Threading;
+using SixLabors.ImageSharp.IO;
+
+namespace SixLabors.ImageSharp.Tests.TestUtilities
+{
+    public class AsyncLocalSwitchableFilesystem : IFileSystem
```

Comment:
By altering local `Configuration` only, this becomes unnecessary, we'll need `SingleStreamFileSystem` only.

Reply:
That means we don't test the code that make sue of the global configuration however

Reply:
makes sense (changed my mind based on what we are actually trying to test)

Reply:
I'd manage this as following:
1. We need a test that somehow checks if the Default configuration is forwarded in all the parameterless `Load` can be  be independent of the individual methods. This is likely a gap, but can be easily fixed by adding `Assert.Same(Configuration.Default, img.GetConfiguration());` to the following method:
https://github.com/SixLabors/ImageSharp/blob/841f8965bedc349d2cf0b8217451d27d054b7ca7/tests/ImageSharp.Tests/Image/ImageTests.Load_FileSystemPath_UseDefaultConfiguration.cs#L20-L23
2. We can test individual features with local `Configuration` when necessary

Otherwise we would have to move the test case into a separate class and do magic with `[CollectionDefinition(DisableParallelization = true)]` (then reset the state change), or use `RemoteExecutor` to separate at process level. I think sticking with isolated local state fits Xunit best practices better.

I'm actually surprised that other (independent) tests are passing while the shared configuration is using `SingleStreamFileSystem`. Maybe it's because `TestImageProvider` also uses [an isolated configuration](https://github.com/SixLabors/ImageSharp/blob/97dde7fb2f0ac97b5b1f9e6b0555a53d2e270c46/tests/ImageSharp.Tests/TestUtilities/ImageProviders/TestImageProvider.cs#L28) and there is a very low amount of tests that will call the shared Configuration's filesystem.

Reply:
not that I matters anymore as I've switched to local configuration, but the reason the other tests where working fine is becasue they where not using the `SingleStreamFileSystem` the singleton `AsyncLocalSwitchableFilesystem` used an `AsyncLocal<T>`  to store the the `SingleStreamFileSytem` and as that was only being overridden in the test(s) the called `ConfigureFileSystemStream()` then all other tests where falling back to proxying to an `LocalFileSytem` which is our default version.